X509.pm

```diff
@@ -180,6 +186,30 @@ sub Crypt::OpenSSL::X509::subjectaltname {
   my $san = $asn_node->decode($bin_data)
       or die 'Unable to decode SubjectAltName: '.$asn_node->error;
 
+  foreach my $name ( @$san ) {
+      foreach my $item (keys %$name) {
+          if( $item eq 'iPAddress' ) {
+              my $ip = $name->{$item};
+              if( length $ip == 4 ) {
+                  $name->{iPAddress} = sprintf( '%d.%d.%d.%d', unpack( 'C4', $ip ) );
+              } elsif( length $ip == 16 ) {
+                  $name->{iPAddress} = sprintf( '%x:%x:%x:%x:%x:%x:%x:%x', unpack( 'n8', $ip ) );
+              } else {
```

Comment:
As noted separately: I don't think it's a good idea to replace the binary with text.  

It would have been the right thing at day 1, but the binary has been there for some time.  I, for one, have code that deals with it.  So this is a breaking change.  For me, no big deal, but we don't know what other code might be out there.

Adding the _txt key in my original patch ensured that it wouldn't break any existing code, but still provides a clean interface for humans (and for most other Perl libraries, that want text.)  I used _txt so the key could be referenced without quoting it.

It seems like the cleanest solution given where we are today.  It's consistent with the way everything else works - you pick the key that you want. And it doesn't break anything.  

I don't see any advantage to adding an 'as_text' function - that would be one more step for the consumer.

Reply:
Then I would be tempted to add a parameter that could be used with subjectaltname to specify whether to add the extra _txt fields or replace the fields with parsed values.

Reply:
Why?  I don't see what problem a parameter solves.

Seems like it would just add more complication in the API.  The new keys are harmless - no existing code should notice them, it should just access the old keys.  Any new code would use the new keys, unless it did something odd (like feed them to the `*sockaddr_in` functions).

Requiring an option to get the old behavior still breaks old code.  Requiring an option to get the new (txt) keys seems unnecessary since they shouldn't bother anyone.  (I suppose there's the case of code that assumes that there's only one key in the hash - but I don't have any sympathy for that.)

You also have to test both cases of the selector - as written, your test only needs to verify that both values are correct.

And, should anyone want both, a selector would require two calls.  Perhaps where you do pass a value to a sockaddr (binary) and log it (text).  

What am I missing?